lib/portage/dbapi/bintree.py

```diff
@@ -1371,6 +1505,7 @@ def _pkgindex_entry(self, cpv):
 		"""
 
 		pkg_path = self.getname(cpv)
+		binpkg_format = self.settings.get("BINPKG_FORMAT", "xpak")
```

Comment:
We can't assume a constant binpkg_format here, since the `_pkgindex_entry` can be called for an arbitrary existing package. So, we'll have to either get the binpkg_format from the cpv parameter somehow, or else pass it in as a new `_pkgindex_entry` parameter.

Reply:
One of the _pkgindex_entry calls to be concerned about it in lib/portage/emaint/modules/binhost/binhost.py (implements `emaint --fix  binhost`). The cpv that is passes in comes from the cpv_all method. If that cpv has a `cpv._metadata['BINPKG_FORMAT']`, then we can use it.

Reply:
I added try use _metadata first, if it not exists fallback to local settings.